plugins/metalsmith-hierarchy.js

```diff
@@ -18,6 +18,26 @@ const firstParagraph = (file) => {
   return contents[0] || "";
 };
 
+/**
+ * Controls wether a file should be build based on environment variables and file annotations
+ */
+const shouldExcludeFile = (file) => {
+  if (
```

Comment:
I think this is a bit easier to understand than a huge logical expression so I decided to have single ones for each environment.